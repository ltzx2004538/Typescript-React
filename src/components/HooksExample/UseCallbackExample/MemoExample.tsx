import React from 'react';

interface MemoProps {
	increment: (number:number) => void;
}

export const MemoExample= React.memo<MemoProps>(
	(props) => {
		const {increment} = props
		console.log('render')
		return (
			<button onClick= {e=> increment(5)}>
				Memo
			</button>
		 );
	}
	);