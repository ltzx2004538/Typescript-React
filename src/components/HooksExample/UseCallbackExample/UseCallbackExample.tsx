import React,{useState, useCallback} from 'react'
import {MemoExample} from './MemoExample';
interface Props {
	testLabel?: string
}

 const UseCallbackExample: React.FC<Props> = (props)=>{
	const {testLabel} = props;
	const [count,setCount] = useState<number>(0)

	const increment = useCallback((number) =>{
		setCount(prevCount => prevCount + number);
	},[setCount])
	 
	return (
		<div>
			<span>
				{count}
			</span>
			<MemoExample increment={increment}/>
			<span data-testid="testLabel">
				Hi {testLabel?testLabel:'no data'}
			</span>
		</div>
	)
}

export default UseCallbackExample;

