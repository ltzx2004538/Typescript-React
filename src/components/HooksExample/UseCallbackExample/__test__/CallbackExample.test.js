import React from 'react';
import ReactDOM from 'react-dom';
import UseCallbackExample from '../UseCallbackExample';
import {render, cleanup} from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom'

afterEach(cleanup);

it("render without crashing",()=>{
	const div = document.createElement("div");
	ReactDOM.render(<UseCallbackExample></UseCallbackExample>,div)
})

// it("renders component correctly",()=> {
// 	const {getByTestId} = render(<UseCallbackExample testLabel={'test1'}></UseCallbackExample>);
// 	expect(getByTestId('testLabel').textContent).toBe('test1');
// })

// it("renders component correctly",()=> {
// 	const {getByTestId} = render(<UseCallbackExample testLabel={'test2'}></UseCallbackExample>);
// 	expect(getByTestId('testLabel').textContent).toBe('test2');
// })

it("matches snapshot 1", ()=>{
	const tree = TestRenderer.create(<UseCallbackExample testLabel={'test1'}></UseCallbackExample>).toJSON();
	expect(tree).toMatchSnapshot();
})

it("matches snapshot 2", ()=>{
	const tree = TestRenderer.create(<UseCallbackExample testLabel={'test2'}></UseCallbackExample>).toJSON();
	expect(tree).toMatchSnapshot();
})