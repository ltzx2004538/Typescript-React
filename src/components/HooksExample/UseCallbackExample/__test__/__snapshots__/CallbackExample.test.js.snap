// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Memo
  </button>
  <span
    data-testid="testLabel"
  >
    Hi 
    test1
  </span>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Memo
  </button>
  <span
    data-testid="testLabel"
  >
    Hi 
    test2
  </span>
</div>
`;
